### PR TITLE
r/function_app: support for User Assigned Identities

### LINKED ISSUE #4607: `azurerm_function_app` - Add support for UserAssigned identity
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Add support for UserAssigned identity for function apps.

### New or Affected Resource(s)

* azurerm_function_app

### Potential Terraform Configuration

```hcl
  identity {
    type         = "UserAssigned"
    identity_ids = []
  }
```

### References

* #0000

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Porting-over the work done by @joakimhew in #4687 before the refactor]

Fixes #4607

### CODE REVIEW

File: azurerm/internal/services/web/resource_arm_function_app.go [hunk 1]
```diff
@@ -423,19 +398,19 @@ func resourceArmFunctionAppUpdate(d *schema.ResourceData, meta interface{}) erro
...
+	if _, ok := d.GetOk("identity"); ok {
+		appServiceIdentityRaw := d.Get("identity").([]interface{})
+		appServiceIdentity := azure.ExpandAppServiceIdentity(appServiceIdentityRaw)
+		siteEnvelope.Identity = appServiceIdentity
 	}
 
 	future, err := client.CreateOrUpdate(ctx, resGroup, name, siteEnvelope)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error updating Function App %q (Resource Group %q): %+v", name, resGroup, err)
```
Comment (jackofallops) [inline azurerm/internal/services/web/resource_arm_function_app.go]: Worth updating this to resourceGroup as elsewhere in the provider while we're here?

File: website/docs/r/function_app.html.markdown [hunk 2]
```diff
@@ -164,9 +164,13 @@ A `cors` block supports the following:
 
 ---
 
-`identity` supports the following:
+A `identity` block supports the following:
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown]:
```suggestion
An `identity` block supports the following:
```

File: azurerm/internal/services/web/resource_arm_function_app.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/internal/services/web/resource_arm_function_app.go]: we'll be switching this out when the Resource ID parsers come through - so we may as well hold off and do this then tbh

**Comment by tombuildsstuff** [comment]
Tests look as expected:

Screenshot 2020-02-11 at 14 32 52

**Pull request merged**